From 148731dc5fc0bed1b7b8629f1043ab028099d289 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@serpentos.com>
Date: Mon, 9 Dec 2024 23:12:20 +0000
Subject: [PATCH 6/6] 50plymouth: Generate proper OS plymouth for generic
 initrd

We use these during kernel builds to ensure we have reproducible
initrds in client installs rather than relying on client side
rebuilds which are costly and prone to errors caused by local
configuration changes/drift.

Signed-off-by: Ikey Doherty <ikey@serpentos.com>
---
 .../50plymouth/plymouth-populate-initrd.sh    | 44 +++++++------------
 1 file changed, 15 insertions(+), 29 deletions(-)

diff --git a/modules.d/50plymouth/plymouth-populate-initrd.sh b/modules.d/50plymouth/plymouth-populate-initrd.sh
index 7e3afdd..6eb51ba 100755
--- a/modules.d/50plymouth/plymouth-populate-initrd.sh
+++ b/modules.d/50plymouth/plymouth-populate-initrd.sh
@@ -12,34 +12,20 @@ mkdir -m 0755 -p "${initdir}/usr/share/plymouth"
 
 inst_libdir_file "plymouth/text.so" "plymouth/details.so"
 
-if [[ $hostonly ]]; then
-    inst_multiple \
-        "/usr/share/plymouth/themes/details/details.plymouth" \
-        "/usr/share/plymouth/themes/text/text.plymouth"
-
-    if [[ -d $dracutsysrootdir/usr/share/plymouth/themes/${PLYMOUTH_THEME} ]]; then
-        for x in "/usr/share/plymouth/themes/${PLYMOUTH_THEME}"/*; do
-            [[ -f "$dracutsysrootdir$x" ]] || break
-            inst "$x"
-        done
-    fi
-
-    if [[ -L $dracutsysrootdir/usr/share/plymouth/themes/default.plymouth ]]; then
-        inst /usr/share/plymouth/themes/default.plymouth
-        # Install plugin for this theme
-        PLYMOUTH_PLUGIN=$(grep "^ModuleName=" "$dracutsysrootdir"/usr/share/plymouth/themes/default.plymouth | while read -r _ b _ || [ -n "$b" ]; do echo "$b"; done)
-        inst_libdir_file "plymouth/${PLYMOUTH_PLUGIN}.so"
-    fi
-else
-    for x in "$dracutsysrootdir"/usr/share/plymouth/themes/{text,details}/*; do
-        [[ -f $x ]] || continue
-        THEME_DIR=$(dirname "${x#"$dracutsysrootdir"}")
-        # shellcheck disable=SC2174
-        mkdir -m 0755 -p "${initdir}/$THEME_DIR"
-        inst_multiple "${x#"$dracutsysrootdir"}"
+inst_multiple \
+    "/usr/share/plymouth/themes/details/details.plymouth" \
+    "/usr/share/plymouth/themes/text/text.plymouth"
+
+if [[ -d $dracutsysrootdir/usr/share/plymouth/themes/${PLYMOUTH_THEME} ]]; then
+    for x in "/usr/share/plymouth/themes/${PLYMOUTH_THEME}"/*; do
+        [[ -f "$dracutsysrootdir$x" ]] || break
+        inst "$x"
     done
-    (
-        cd "${initdir}"/usr/share/plymouth/themes || exit
-        ln -s text/text.plymouth default.plymouth 2>&1
-    )
+fi
+
+if [[ -L $dracutsysrootdir/usr/share/plymouth/themes/default.plymouth ]]; then
+    inst /usr/share/plymouth/themes/default.plymouth
+    # Install plugin for this theme
+    PLYMOUTH_PLUGIN=$(grep "^ModuleName=" "$dracutsysrootdir"/usr/share/plymouth/themes/default.plymouth | while read -r _ b _ || [ -n "$b" ]; do echo "$b"; done)
+    inst_libdir_file "plymouth/${PLYMOUTH_PLUGIN}.so"
 fi
-- 
2.47.1

